templates/article.html

```diff
@@ -5,6 +5,13 @@
 <meta name="description" content="{{ article.summary|striptags|escape }}" />
 <meta name="keywords" content="{{ article.tags|join(', ')|escape }}">
 
+{% if article.translations -%}
+<link rel="alternate" href="{{ SITEURL }}/{{ article.url }}" hreflang="{{ article.lang }}" />
```

Comment:
Should be canonical no? Or which one SEO will consider as the main URL?

Reply:
I can add a canonical link pointing to the main language version, if it exists. If it doesn't we won't have a way of knowing which one is the canonical one (custom `Canonical` attribute in the document?)

SEO is a complex world, I found these articles about canonical and alternate:
https://yoast.com/rel-canonical/
https://moz.com/community/q/rel-canonical-and-rel-alternate-both-necessary

Reply:
I know that SEO is complex, that's why a few things are missing.

One thing is true about Pelican, there is a default language, thus that should be the canonical one. If you set your site to EN-US, the canonical should redirect to the post/page in English, I'm not sure if it's possible to have EN website with actual no post in EN.

It's trick. Missing canonical link does harm SEO or not? If not we can skip it. I do remember that canonical should be included, but not sure if it's required.

Reply:
Can we actually get the `DEFAULT_LANG` in a reliable way? Because the *i18n_subsites* plugin overrides that when rendering each subsite. From their docs:

> DEFAULT_LANG
> This should not be overridden as the plugin changes it to the language code of each sub-site to change what is perceived as translations.

I'm all for improving SEO, but my suggestion would be to merge the alternate links, because they shouldn't hurt, and figure out the canonical stuff later.

Reply:
Sounds good.